sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -489,7 +491,8 @@ def big_data_passing_tasks(prname: str, task: dict, pipelinerun_template: dict,
     # Data passing for the task outputs
     appended_taskrun_name = False
     for task_output in task.get('taskSpec', {}).get('results', []):
-        if (task_name, task_output.get('name')) in outputs_tasks:
+        if (task_name, task_output.get('name')) in outputs_tasks or \
+            task_spec.get('metadata', {}).get('labels', {}).get(ARTIFACT_OUTPUT_LABEL_KEY, '') == ARTIFACT_OUTPUT_LABEL_VALUE:
```

Comment:
So it looks like the "big data" mode can only be turned-on for the whole component, i.e. all of its outputs?

It doesn't sound like a good idea. It's quite typical for some component to return both small outputs (such as status message) and big outputs (such as standard output of the operation being run).

Reply:
For big data outputs, we will copy all the outputs into the object storage and only put back the results that are under 3kb.

Reply:
If you want to only output a particular output, I can refactor the current big data passing logic. But in that case I will need the task output names as part of the label values.

Reply:
Sure, that will do.

The idea you've described is interesting too, but we'd like to have more control over the mechanism, possibly do what you're describing "manually" in a dedicated controller (may improve both performance and error-handling).

Reply:
I converted the flag to an annotation that contain a list of output names that will be considered as artifact. You can checkout the test case when `data2` is not specified in the annotation, `data2` will get treated as regular parameter output.